database/migrations/2017_02_04_025727_add_link_table_indexes.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class AddLinkTableIndexes extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('links', function (Blueprint $table)
+        {
+            // Add long_url hashes
+            $table->unique('short_url');
+            $table->string('long_url_hash', 10)->nullable();
+            $table->index('long_url_hash', 'links_long_url_index');
+        });
+
+        DB::statement("UPDATE links SET long_url_hash = crc32(long_url);");
```

Comment:
There might be a non-SQL way of adding crc32 hashes on existing links without causing problems with databases such as Postgres. Postgres doesn't seem to have crc32 built-in